File: app/src/main/java/com/jjl/sudokuplay/SudokuRvAdapter.java
package com.jjl.sudokuplay;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class SudokuRvAdapter extends RecyclerView.Adapter {

    private Context mContext;
    private List<Integer> mList;
    private final LayoutInflater mLocalinflater;

    public SudokuRvAdapter(Context context, List<Integer> list) {
        mContext = context;
        mList = list;
        mLocalinflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new SudokuViewHolder(mLocalinflater.inflate(R.layout.item_sudoku ,viewGroup , false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int index) {
        ((SudokuViewHolder)viewHolder).setData(mList , index);
    }

    @Override
    public int getItemCount() {
        if(mList != null)
            return mList.size();
        return 0;
    }

    class SudokuViewHolder extends RecyclerView.ViewHolder{

        private TextView mTextView;

        public SudokuViewHolder(@NonNull View itemView) {
            super(itemView);
            mTextView = itemView.findViewById(R.id.textView);
        }

        public void  setData( List<Integer> data , int index){
            mTextView.setText(String.valueOf(data.get(index)));
        }
    }
}

File: app/src/main/java/com/jjl/sudokuplay/SudokuRvAdapter.kt
package com.jjl.sudokuplay

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

class SudokuRvAdapter(private val mContext: Context, private val mList: List<Int>?) : RecyclerView.Adapter<SudokuRvAdapter.SudokuViewHolder>() {
    private val mLocalinflater: LayoutInflater

    init {
        mLocalinflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): SudokuViewHolder {
        return SudokuViewHolder(mLocalinflater.inflate(R.layout.item_sudoku, viewGroup, false))
    }

    override fun onBindViewHolder(viewHolder: SudokuViewHolder, index: Int) {
        viewHolder.setData(mList, index)
    }

    override fun getItemCount(): Int {
        return mList?.size ?: 0
    }

    inner class SudokuViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val mTextView: TextView

        init {
            mTextView = itemView.findViewById(R.id.textView)
        }

        fun setData(data: List<Int>?, index: Int) {
            mTextView.text = data?.get(index).toString()
        }
    }

    fun Any?.toString(): String {
        if (this == null) return "null"
        return toString()
    }
}
